test/vite-plugin-filter-replace.mts

```diff
@@ -0,0 +1,184 @@
+/**
+ * Taken from https://github.com/ikeq/vite-plugin-filter-replace/tree/main
+ *
+ * Modified to work with vitest browser @v3+
```

Comment:
Nitpick: Maybe this should reside in `@mui/internal-test-utils`? 🤔

Reply:
Possibly, but I would rather not right now 😅

Reply:
Sure, makes sense. 😆 
We can move it if it's needed in the `material-ui` or `base-ui` repo. 👌